crates/c_api/macro/lib.rs

```diff
@@ -150,3 +158,39 @@ pub fn declare_ref(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
     })
     .into()
 }
+
+#[proc_macro_attribute]
+pub fn prefix_symbol(
+    _attributes: proc_macro::TokenStream,
+    input: proc_macro::TokenStream,
+) -> proc_macro::TokenStream {
+    let mut stream = TokenStream::from(input.clone()).into_iter();
+
+    let mut fn_name = None;
+
+    while let Some(token) = stream.next() {
+        match token {
+            TokenTree::Ident(ref ident) if ident.to_string() == "fn" => {
+                if let Some(TokenTree::Ident(ident_name)) = stream.next() {
+                    fn_name = Some(ident_name.to_string());
+                    break;
+                }
+            }
+            _ => continue,
+        }
+    }
+
+    if fn_name.is_none() {
+        panic!("expected a valid Rust function definition, but it does not appear in: {input:?}");
+    }
+
+    let prefixed_fn_name = format!("wasmi_{}", fn_name.unwrap());
+
+    let mut attr: proc_macro::TokenStream = quote! {
+        #[cfg_attr(feature = "prefix-symbols", export_name = #prefixed_fn_name)]
+    }
+    .into();
+    attr.extend(input);
+
+    attr
+}
```

Comment:
Thank you! I think this is fine for merging even though it does not really check for function signatures. What I had in my mind is to do parsing similar to `syn` but overly specific to our use case as to not have to use `syn` as dependency. Shouldn't be too much work but I am willing to implement this as a follow-up PR so we can get this merged.